onnx/common/stl_backports.h

```diff
@@ -2,8 +2,62 @@
 
 // This file contains backports of STL features for newer C++.
 
+#include <memory>
+#include <type_traits>
+
+/*
+ * Use MOVE_CAPTURE_IF_CPP14 in a lambda capture so it gets
+ * copied in C++11 and moved in C++14.
+ * Example:
+ *   std::string mystring;
+ *   auto lambda = [MOVE_CAPTURE_IF_CPP14(mystring)] {
+ *     std::cout << mystring;
+ *   }
+ */
 #ifdef __cpp_init_captures
   #define MOVE_CAPTURE_IF_CPP14(variable) variable = std::move(variable)
 #else
   #define MOVE_CAPTURE_IF_CPP14(variable) variable
 #endif
+
+
+/**
+ * For exception safety and consistency with make_shared. Erase me when
+ * we have std::make_unique().
+ *
+ * @author Louis Brandy (ldbrandy@fb.com)
+ * @author Xu Ning (xning@fb.com)
+ * @author Sebastian Messmer (messmer@fb.com)
+ */
+
+#if __cplusplus >= 201402L || __cpp_lib_make_unique >= 201304L || \
+    (__ANDROID__ && __cplusplus >= 201300L) || _MSC_VER >= 1900
+
+// std::make_unique is available
+
+#else
+
+namespace std {
+
+template <typename T, typename... Args>
+typename std::enable_if<!std::is_array<T>::value, std::unique_ptr<T>>::type
+make_unique(Args&&... args) {
```

Comment:
Let's put it inside `ONNX_NAMESPACE` (In case of either our detection macros is not robust enough or there is another library does the same trick to pollute std namespace, there will be multiple definitions of std::make_unique)